gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -784,7 +784,12 @@ protected GoogleCloudStorageItemInfo getInitialMetadata() {
 
       @Override
       protected Storage.Objects.Get createRequest() throws IOException {
-        return initializeRequest(super.createRequest(), resourceId.getBucketName());
+        // Creates a storage request that will be used to request media data from GCS.
+        // Sets "alt=media" query parameter to indicate that this is a media request (instead of
+        // a metadata request) so that the request can be initialized properly.
+        Storage.Objects.Get request = super.createRequest();
+        request.set("alt", "media");
```

Comment:
I do not think that all requests created by this method are media requests. Also, why do we need to set this manually, I thought it will be auto-set by client library depending on the request?

Reply:
I'll read the code again to confirm the first question. To the second question, we need the `alt=media` parameter for `initializeRequest` to distinguish the meaning of the request. The parameter is only set in `executeMedia()` function, and we it's too late for access token initializing. Unfortunately this means that we can't guarantee that every requests that will eventually call `executeMedia` is initialized with the correct access token.

Reply:
The only call I'm not sure about is [here](https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/master/gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageReadChannel.java#L234). The comment says it is supposed to get metadata but it calls `execute` instead of `executeMedia`.

Reply:
In this case I think that we may want something more robust/future proof then setting manually query parameter, that can be overwritten anyway?

One option would be to extend `Get` class with `GetData` and `GetMetdata` that override `execute`/`executeMedia` requests to fail in cases that they are not supported - this will allow to know how request will be executed as soon as its object created.

Reply:
Although this doesn't prevent future codes from using the base class directly, I think it's a good start to decouple the actions.

Also, regarding to the code linked above ([this](https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/master/gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageReadChannel.java#L234). Do you know the intent of this call? I can change it and run a few beeline/spark jobs to see if it works, but it would be better if we can for sure know that this is meant to be a metadata call instead.

Reply:
Appended a new commit for this change.